dvc/remote/local/slow_link_detection.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import time
+
+from dvc.config import Config
+from dvc.utils import boxify
+
+logger = logging.getLogger(__name__)
+
+
+class SlowLinkDetectorDecorator(object):
+    LINKING_TIME_SECONDS = 20.0
+    SHOULD_DISPLAY = True
+
+    def __init__(self, method):
+        self.method = method
+
+    def __call__(self, remote_local, *args, **kwargs):
```

Comment:
does it actually need to know `remote_local`? Or should it simply be `(self, *args, **kwargs)`?

Reply:
it can be self, I just named it to preserve information what user can expect here

Reply:
@pared not sure I follow. `remote_local` is not used, you are just passing it through, aren't you? If so, your `__call__` doesn't care about it.

Reply:
Oh, you are right, id matters only for guard, not for decorator class.